addons/l10n_nz/data/account_tax_report_data.xml

```diff
@@ -85,29 +94,59 @@
                         <field name="name">[BOX 12] Multiply BOX11 by 3 and then divide by 23</field>
                         <field name="code">NZBOX12</field>
+                        <field name="sequence">220</field>
                         <field name="aggregation_formula">(NZBOX11.balance * 3)/23</field>
                     </record>
                     <record id="tax_report_box13" model="account.report.line">
                         <field name="name">[BOX 13] Credit adjustments from your calculation sheet</field>
                         <field name="code">NZBOX13</field>
+                        <field name="sequence">230</field>
                         <field name="expression_ids">
-                            <record id="tax_report_box13_formula" model="account.report.expression">
+                            <record id="tax_report_box13_aggregation" model="account.report.expression">
                                 <field name="label">balance</field>
-                                <field name="engine">external</field>
-                                <field name="formula">sum</field>
-                                <field name="subformula">editable;rounding=2</field>
+                                <field name="engine">aggregation</field>
+                                <field name="formula">NZBOX13_tags.balance + NZBOX13_manual.balance</field>
+                            </record>
+                        </field>
+                        <field name="children_ids">
+                            <record id="tax_report_box13_tags" model="account.report.line">
+                                <field name="name">[BOX 13] Amount computed from tax tags</field>
+                                <field name="code">NZBOX13_tags</field>
+                                <field name="sequence">233</field>
+                                <field name="expression_ids">
+                                    <record id="tax_report_box13_tag" model="account.report.expression">
+                                        <field name="label">balance</field>
+                                        <field name="engine">tax_tags</field>
+                                        <field name="formula">BOX 13</field>
+                                    </record>
+                                </field>
+                            </record>
+                            <record id="tax_report_box13_manual" model="account.report.line">
+                                <field name="name">[BOX 13] Manual adjustments</field>
+                                <field name="code">NZBOX13_manual</field>
+                                <field name="sequence">236</field>
+                                <field name="expression_ids">
+                                    <record id="tax_report_box13_formula" model="account.report.expression">
+                                        <field name="label">balance</field>
```

Comment:
I think we may need a local migration script after all :smile: 
In order to avoid triggering the `account_report_expression_line_label_uniq` constraint, we could simply have a `pre-` script that would change this expression's `label` to anything other than `balance`.
That should solve the upgrade error we currently have.

Reply:
Wouldn't we lose the balance that's used for the report ? The error comes from the fact it tries to upgrade to a newer version that doesn't have this fix, and thus tries to create a new account report expression with balance, I don't know if the related manual values will be kept if we do it this way... I thought that's why we needed either the exception override (and merge the fw-port fast) or to start with master and chain-backport

Reply:
No, we would not lose any data, since the expression `label` would be reset to `balance` during the update.
But we would not trigger the constraint since the `label` would not be `balance` at that point, thanks to the `pre-` script.

Reply:
I'll try that then

Reply:
That means I'll have to copy-paste that script for every versions (17.0, saas-17.4, 18.0, saas-18.1, saas-18.2, saas-18.3) ? Wouldn't it be cleaner and easier to override the exception and be quick in the forward-porting ?

Reply:
I think overriding the exception would not work.
The fact that the SQL constraint is triggered means the write operation is rolled back, along with the values that were expected to be updated.
But I think won't have to duplicate the script.
You can put it in a `0.0.0` folder and it will be executed for all versions (see [here](https://github.com/odoo/upgrade/wiki/How-To#where-to-write-upgrade-scripts)).

Reply:
> Moreover, I would need to add a condition to the script to only apply until saas-18.3 (to avoid future versions to unnecessarily modify the expressions, as master would already have the fix).

In master the script will never run since upgrades 18->master will have the script in the latest version.

Reply:
> That means I'll have to copy-paste that script for every versions (17.0, saas-17.4, 18.0, saas-18.1, saas-18.2, saas-18.3) ? Wouldn't it be cleaner and easier to override the exception and be quick in the forward-porting ?

No. It means that you forward ports all will have the local script. Which is **correct**. Because clients will need to upgrade 1.1->1.2 in all released versions. Your forward ports will ensure that all those versions get the script. You can even leave it in master for history. No DB that started from 18.3 will ever run the script since they would have gotten 1.2 since day one, thus they won't never need to upgrade it.

Reply:
But you are right in one thing: runbot cannot deal with this :/ So backport it must be :/